### PR TITLE
Stencilbook added to program-tools

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> A clone of Storybook, but built with Stencil. Used to provide an easy to view/test implemented components and all of their states / specs, and to easily share progress with customers.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-2993
 - Process.st launch checklist: https://app.process.st/checklists/Stencilbook-10-qWvgtXHd-muYcH46ah1F-w

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/stencilbook/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@saasquatch/stencilbook",
+  "version": "0.0.1-beta6",
```
Comment (00salmon) [inline packages/stencilbook/package.json:3]:
```suggestion
  "version": "1.0.0",
```

File: packages/stencilbook/src/stencil-storybook.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                      <li class="parentStory">
+                        <details style={{ marginBottom: "10px" }}>
+                          <summary style={{ outline: "none" }}>
+                            {s.story.title}
+                          </summary>
+                          {s.subs &&
+                            Object.keys(s.subs).map((subKey) => {
+                              const key = group + "/" + subKey;
+                              const subStory = s.subs[subKey];
+                              const subStoryView = () => <subStory />;
```
Comment (00salmon) [inline packages/stencilbook/src/stencil-storybook.tsx:178]:
```suggestion
                              const subStoryView = () => <subStory />;
```
this is never used

File: packages/stencilbook/src/stencil-storybook.tsx [hunk 3]
```diff
@@ -0,0 +1,226 @@
+import { h as StencilH, FunctionalComponent, VNode } from "@stencil/core";
+import * as Hooks from "@saasquatch/stencil-hooks";
```
Comment (loganvolkers) [inline packages/stencilbook/src/stencil-storybook.tsx:2]: Should swap to universal hooks

```suggestion
import * as Hooks from "@saasquatch/stencil-hooks";
```

**Review by loganvolkers — CHANGES_REQUESTED** [review]
Switch to universal hooks

**Comment by noahwc** [comment]
Added custom homepage with a new `home` prop. Stencilbook initially loads to the homepage and users can return to the homepage by clicking on the header.

File: packages/stencilbook/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  "exports": "./dist/index.modern.js",
+  "module": "dist/index.module.js",
+  "scripts": {
+    "build": "microbundle",
+    "dev": "microbundle watch",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "dependencies": {
+    "@emotion/css": "^11.1.3",
+    "@saasquatch/stencil-hooks": "^2.0.0",
```
Comment (loganvolkers) [inline packages/stencilbook/package.json:16]: Not required anymore?

```suggestion
```

**Review by loganvolkers — CHANGES_REQUESTED** [review]
@00salmon Add a changelog

**Review by loganvolkers — APPROVED** [review]
Looks good to me!

**Pull request merged**